### DISCUSSION TITLE
Is there a way to configure Tasmota fallback topic?

### DISCUSSION THREAD

**Discussion by ramanraja** [arendst/Tasmota#13725] [Q&A] [answered]
I want a configurable fallback topic for my Tasmota devices.
My situation is this:
I have a number of users, in different locations. Each user can have multiple Tasmota devices. They all connect to the same public MQTT broker provided by Amazon. Their access to the MQTT topic is secured through AWS policies.
My Tasmota topic is of the format 
 `cmnd/USER_ID/DEVICE_ID/POWER`
I have a secure login page in my app where I collect the user id; the backend automatically injects the USER_ID into the command before sending it to Tasmota.
Using this mechanism, a user can control only devices owned by him/her. Spoofing another user's ID is not possible since the '`USER_ID`' substring on the topic is managed only by the server.
Now Tasmota's fallback topic is a threat to this security model :) 
Tasmota has a fixed fallback topic format, so anyone who knows your device MAC address can control it remotely.
Configuring a fallback topic of the form  `cmnd/USER_ID/DVES_xxxx_fb`  will solve this problem.
Is there any way (including modifying the source code) to configure the fallback topic ? In the worst case, I am willing to disable the fallback topic altogether.

**Comment by sfromis** [comment, marked as answer]
If you configure MqttClient (not to the same as Topic), this will also affect fallback topic.

**Reply by ramanraja** [reply]
Thanks. This is such a useful tip, I want to elaborate the  solution for my own, and others'  future reference:
Originally my Information page showed:
```
MQTT Client	DVES_709D52
Fallback Topic	cmnd/DVES_709D52_fb/
```
The only way to change the fallback topic is to change the MQTT Client ID.
So I issued the command:  `MqttClient  USER_999999/DVES_%06X_fb`
Now my info page shows:
```
MQTT Client	USER_999999/DVES_709D52
Fallback Topic	cmnd/USER_999999/DVES_709D52_fb/
```
Thank you !
